packages/gatsby/src/utils/websocket-manager.js

```diff
@@ -45,17 +45,19 @@ class WebsocketManager {
 
       const leaveRoom = path => {
         s.leave(getRoomNameFromPath(path))
+        const leftRoom = this.websocket.sockets.adapter.rooms[
+          getRoomNameFromPath(path)
+        ]
+        if (!leftRoom || leftRoom.length === 0) {
+          this.activePaths.delete(path)
+        }
         activePath = null
       }
 
-      s.on(`disconnect`, s => {
-        leaveRoom(activePath)
-      })
-
```

Comment:
Just checking here - I think after removing this handler, when browser tab will be closed (so `unregisterPath` event won't be emitted), active path won't get unregistered and therefore query for that path will continue to be prioritized

Reply:
Yeah... that's right — lemme add this back real quick